### PR TITLE
PSP: Rename `restricted` to `zz-minimal`

### PR DESCRIPTION
**Pull request by surajssd**
* This PR renames the PSP `restricted` to `zz-minimal`, so that it is the last one in the list of PSPs but before the `privileged` PSP. `restricted` is applicable to all the workloads in the cluster. If a component already ships a PSP then that specific PSP should be the one that is applied to the workload, not the `restricted` PSP. So keeping it last in the list makes sure that pod is not filtered under `restricted` PSP. This was discovered when `restricted` PSP was interfering with the `rook` PSP.

* This PR renames the PSP `privileged` to `zz-privileged`. This is done in order to avoid the situation where `privileged` PSP is picked up accidently even though a specific, small scoped (in terms of permissions) PSP exists.

* openebs: Rename PSP usage. Use policy `zz-privileged` and not `privileged`.

### CODE REVIEW

File: assets/lokomotive-kubernetes/bootkube/resources/charts/kubernetes/templates/psp-restricted.yaml [hunk 1]
```diff
@@ -50,21 +50,21 @@ spec:
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: restricted-psp
+  name: zz-restricted-psp
```
Comment (invidian) [inline assets/lokomotive-kubernetes/bootkube/resources/charts/kubernetes/templates/psp-restricted.yaml:53]: Not sure if it make sense to rename the role. I'd keep it the same as everything else and only change the PSP object name, as it has functional meaning.

Comment (surajssd) [reply assets/lokomotive-kubernetes/bootkube/resources/charts/kubernetes/templates/psp-restricted.yaml:53]: Yeah I wondering what is the right thing to do here, but eventually decided on renaming all the restricted related thing to prefix `zz-`

Comment (surajssd) [reply assets/lokomotive-kubernetes/bootkube/resources/charts/kubernetes/templates/psp-restricted.yaml]: removed it.

**Comment by iaguis** [comment]
CI is failing, can you retrigger?

**Comment by surajssd** [comment]
I have a hunch that openebs fails and am currently testing this issue locally.

**Comment by invidian** [comment]
@surajssd shouldn't `restricted` be named `aa-restricted` or something? So if it's compatible with with some pod, it will be used?

**Comment by surajssd** [comment]
> @surajssd shouldn't restricted be named aa-restricted or something? So if it's compatible with with some pod, it will be used?

No, it will cause problems. So right now any workload in the cluster has at least one policy that it can go through i.e. `restricted` PSP. And it happens because we have this binding here: https://github.com/kinvolk/lokomotive/blob/9bd75f062c4b7cd7c828826fffd2afcb8177a8d1/assets/lokomotive-kubernetes/bootkube/resources/charts/kubernetes/templates/psp-restricted.yaml#L60-L72

Now imagine a workload which has it's own PSP. Now that workload is eligible for two PSPs one is its own and another is `restricted` one. Now the problem here is if `restricted` PSP takes precedence(in alphabetical order) for the application then that workload might not work reliably. Because the PSP that was shipped with the component is not applied but `restricted` PSP is applied, which greatly hinders what an app in the pod can do.

So there are two ways to solve this problem either we remove the aforementioned binding: https://github.com/kinvolk/lokomotive/blob/9bd75f062c4b7cd7c828826fffd2afcb8177a8d1/assets/lokomotive-kubernetes/bootkube/resources/charts/kubernetes/templates/psp-restricted.yaml#L69-L72

By removing this binding we withdraw the ability of an application workload to work out of the box in Lokomotive, especially those that don't ship PSP. So we will have to create Rolebinding as needed. That is provide `restricted` PSP on a case by case basis. Like someone has to create a Rolebinding when they create a new namespace as follows:

```yaml
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: restricted-psp-mywebapp
  namespace: mywebapp
roleRef:
  kind: ClusterRole
  name: restricted-psp
  apiGroup: rbac.authorization.k8s.io
subjects:
- kind: ServiceAccount
  name: default
  namespace: mywebapp
```

OR we solve the problem by making sure the ordering is correct in the PSP names like it is done in this PR.

---

Now we have to make a choice to be specific a trade off between usability and security here.

File: assets/lokomotive-kubernetes/bootkube/resources/charts/kubernetes/templates/psp-restricted.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: policy/v1beta1
 kind: PodSecurityPolicy
 metadata:
-  name: restricted
+  name: zz-restricted
```
Comment (invidian) [inline assets/lokomotive-kubernetes/bootkube/resources/charts/kubernetes/templates/psp-restricted.yaml:4]: Maybe we could name it to `zz-minimal`, so then it goes before `zz-privileged`?

**Pull request merged**